### ISSUE TITLE
unicaode, support unfriendly

### ISSUE THREAD

**Issue by nihuai** [x64dbg/x64dbg#1508]
language：Chine
system：win10-x64
Software：x32dbg
-----------------------------------
x32dbg and VS2015（unicaode）
![image](https://cloud.githubusercontent.com/assets/22935890/24320594/72fd86aa-1173-11e7-9837-da0566e819fc.png)

**Comment by gmastergreatee** [comment]
Seems to me a duplicate of #1366 (2nd last checkbox)

**Comment by mrexodia** [comment]
I don't think that issue is related @gmastergreatee

@nihuai the unicode support in that dump is extended ascii only. Rendering Chinese characters is very expensive but you can enable it by using a code page -> utf16 dump.

**Comment by nihuai** [comment]
@mrexodia

**Comment by nihuai** [comment]
![image](https://cloud.githubusercontent.com/assets/22935890/24321934/40c8c98a-1194-11e7-87b0-df4d31bce78e.png)

**Comment by mrexodia** [comment]
That is correct. The unicode support is severely limited because of the way everything is drawn. The problem is where to cut the unicode string. Tell me if you have any suggestions on how to improve this.

**Comment by mrexodia** [comment]
The issue with selection is because of the font. Unicode characters don't have a maximum width from what I know so you cannot align them easily...

**Comment by nihuai** [comment]
@mrexodia I've found that a character in UTF-16 can be selected properly ONLY when the selection is started from the column of odd number and ended next column in the hexadecimal region。I believe it is better if 2 bytes in hexadecimal region can only represent 1 character in UTF-16 mode and therefore, if I've selected 3 bytes, an additional 1 byte would be automatically added into my selection (4 bytes in total), so as to guarantee that 2 characters are selected and 2 bytes in the selection indicate 1 character。if the value represented by the selection does not exceed the limit of ASCII, only 1 character is selected otherwise, 2 characters are selected, since it is likely to be one of some special characters like Chinese

**Comment by nihuai** [comment]
![image](https://cloud.githubusercontent.com/assets/22935890/24323116/f4c71862-11ab-11e7-88ff-8b53e328b7ee.png)
Picture can not upload

**Comment by nihuai** [comment]
![default](https://cloud.githubusercontent.com/assets/22935890/24323358/571c6e24-11af-11e7-888f-c819ae3cba92.PNG)

**Comment by nihuai** [comment]
![2](https://cloud.githubusercontent.com/assets/22935890/24323427/04cb347e-11b0-11e7-8529-fc499b1cad3a.PNG)

**Comment by mrexodia** [comment]
@nihuai 

> I've found that a character in UTF-16 can be selected properly ONLY when the selection is started from the column of odd number and ended next column in the hexadecimal region

Indeed that is the idea. Proper alignment is the responsibility of the user.

While I don't disagree with the rest of your points I don't currently have any plans implementing it. If you really need certain things for unicode support I am willing to merge them in, otherwise you will have to write a plugin or something that exactly does what you need.

**Comment by nihuai** [comment]
@mrexodia Thank you, look forward to that day
